pkg/v1/tarball/image.go

```diff
@@ -223,7 +223,20 @@ func extractFileFromTar(opener Opener, filePath string) (io.ReadCloser, error) {
 			f.Close()
 		}
 	}()
-
+	iscompressed, err := gzip.Is(f)
```

Comment:
This will add an extra read of the file -- can you use `gzip.Peek` for this instead?

Reply:
I will look into it, sorry for the delay as i was away from work